File: 800/C++/p16.cpp
#include<bits/stdc++.h>
using namespace std;

void solve() {
    int n;
    cin>>n;
    vector<long>arr(n);
    for(int i = 0; i < n; i++) cin>>arr[i];

    long mini = LONG_MAX;
    for(int i = 1; i < n; i++) {
        // if already desorted
        if(arr[i] < arr[i-1]) {
            cout<<0<<endl;
            return;
        }

        mini = min(mini, (arr[i] - arr[i-1]) / 2 + 1);
    }

    cout<<mini<<endl;
}

int main() {
    int t;
    cin>>t;
    while(t--)  solve();
    return 0;
}

File: 800/Java/P16.java
import java.util.*;

public class P16 {
    static void solve(Scanner scanner) {
        int n = scanner.nextInt();
        long []arr = new long[n];
        for(int i = 0; i < n; i++)  arr[i] = scanner.nextLong();

        long mini = Long.MAX_VALUE;
        for(int i = 1; i < n; i++) {
            // if already desorted
            if(arr[i] < arr[i-1]) {
                System.out.println(0);
                return;
            }

            long diff = arr[i] - arr[i-1];
            long step = diff / 2 + 1;
            mini = Math.min(mini, step);
        }

        System.out.println(mini);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        while(t-- > 0)  solve(scanner);
        scanner.close();
    }
}
